event/EventCard.tsx

```diff
@@ -32,7 +25,7 @@ export type EventCardProps = {
   style?: StyleProp<ViewStyle>
 }
 
-export const EventCard = ({
+export const EventCard = memo(function EventCard({
```

Comment:
Minor Nit: When dealing with a larger component (>50 LOC), it's a bit more readable (removes a level of nesting) to keep the main component internal to the file, and then to export a call to memo on the internal component.

So:
```ts
const _EventCard = (props: EventCardProps) => {
  // Component Code...
}

export const EventCard = memo(_EventCard)
```